packages/modelviewer.dev/src/docs-and-examples/sidebar.ts

```diff
@@ -205,6 +205,40 @@ function handlePageJump(entries: IntersectionObserverEntry[]) {
   }
 }
 
+let intersectionRatios = new Map<string, number>();
+function handleExamples(entries: IntersectionObserverEntry[], _observer: any) {
+  if (isFirstOpen) {
+    everyEntry = entries;
+    isFirstOpen = false;
+    document.querySelector(`h3[id="active-container-sidebar"`)!.classList.add(
+        'active');
+  }
+
+  for (const entry of entries) {
+    const id = entry.target.getAttribute('id')!;
+    intersectionRatios.set(id, entry.intersectionRatio);
+  }
+
+  let maxRatio = 0;
+  let maxName = '';
+  for (const name of intersectionRatios.keys()) {
+    if (intersectionRatios.get(name)! > maxRatio) {
```

Comment:
if you set `const ratio = intersectionRatios.get(name)!;` you won't need so many bangs, plus you won't have to look things up twice.

Reply:
done